### ISSUE TITLE
Unmet peer dependencies for  @aws-amplify/core, @aws-amplify/geo and react-native

### ISSUE THREAD

**Issue by markedwards** [aws-amplify/amplify-js#10920]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React

### Amplify APIs

Authentication

### Amplify Categories

auth

### Environment information

<details>

```
# Put output below this line

  System:
    OS: Linux 4.14 Debian GNU/Linux 11 (bullseye) 11 (bullseye)
    CPU: (4) x64 AMD EPYC 7571
    Memory: 5.39 GB / 15.52 GB
    Container: Yes
    Shell: Unknown - /bin/sh
  Binaries:
    Node: 18.12.1 - /usr/bin/node
    Yarn: 1.22.19 - /usr/bin/yarn
    npm: 8.19.2 - /usr/bin/npm
  npmPackages:
    axios: ^1.3.0 => 1.3.0 
  npmGlobalPackages:
    corepack: 0.14.2
    npm: 8.19.2
    yarn: 1.22.19

```

</details>

### Describe the bug

When adding aws-amplify with `yarn add @aws-amplify/ui-react aws-amplify`, there are three warnings about missing peer dependencies:
```
@aws-amplify/ui-react > maplibre-gl-js-amplify@3.0.0" has unmet peer dependency "@aws-amplify/core@5.x.x".
@aws-amplify/ui-react > maplibre-gl-js-amplify@3.0.0" has unmet peer dependency "@aws-amplify/geo@2.x.x".
aws-amplify > @aws-amplify/analytics > @aws-sdk/client-firehose > @aws-sdk/middleware-retry > react-native-get-random-values@1.8.0" has unmet peer dependency "react-native@>=0.56".
```

This does not appear to affect functionality in my case, but these warnings should not occur.  I see #9119 which might be related, but doesn't seem to have gone anywhere.

### Expected behavior

These warnings are meaningless and should not happen.

### Reproduction steps

1. `yarn add @aws-amplify/ui-react aws-amplify`
2. `yarn install`
3. observe warnings about missing peer dependencies for `@aws-amplify/core`, `@aws-amplify/geo` and `react-native`

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by reesscot** [comment]
@markedwards You can track the ui-react related warnings here: https://github.com/aws-amplify/amplify-ui/issues/3393

**Comment by markedwards** [comment]
Thanks

**Comment by cwomack** [comment]
@markedwards, we appreciate you opening up this issue!  It looks like we've got fixes to the warnings being tracked in a few areas as outlined below:

`@aws-amplify/ui-react` related warnings in the [UI Issue #3393](https://github.com/aws-amplify/amplify-ui/issues/3393):

- @aws-amplify/ui-react > maplibre-gl-js-amplify@3.0.0" has unmet peer dependency "@aws-amplify/core@5.x.x".
- @aws-amplify/ui-react > maplibre-gl-js-amplify@3.0.0" has unmet peer dependency "@aws-amplify/geo@2.x.x".

The third warning will be taken care of via upcoming upgrades to the SDK:
- aws-amplify > @aws-amplify/analytics > @aws-sdk/client-firehose > @aws-sdk/middleware-retry > react-native-get-random-values@1.8.0" has unmet peer dependency "react-native@>=0.56".

Going to close out this issue after verifying all warnings are being tracked for fixes in other areas so we don't duplicate efforts.

**Comment by stephenh** [comment]
@cwomack when should the SDK upgrades land? AFAIU I'm blocked on upgrading my app to React 18 until #9119  is fixed:

```
npm ERR! Could not resolve dependency:
npm ERR! peer react@"17.0.2" from react-native@0.66.0
npm ERR! node_modules/react-native
npm ERR!   peer react-native@"*" from @react-native-community/cli@6.1.0
npm ERR!   node_modules/@react-native-community/cli
npm ERR!     @react-native-community/cli@"^6.0.0" from react-native@0.66.0
npm ERR!   peer react-native@">=0.56" from react-native-get-random-values@1.7.0
npm ERR!   node_modules/react-native-get-random-values
npm ERR!     react-native-get-random-values@"^1.4.0" from @aws-sdk/middleware-retry@3.6.1
npm ERR!     node_modules/@aws-sdk/middleware-retry
npm ERR!       @aws-sdk/middleware-retry@"3.6.1" from @aws-sdk/client-cloudwatch-logs@3.6.1
npm ERR!       node_modules/@aws-sdk/client-cloudwatch-logs
npm ERR!       11 more (@aws-sdk/client-cognito-identity, ...)
```

**Comment by cwomack** [comment]
@stephenh, there shouldn't be an issue with React 18 and Amplify.  I've got some local projects that have been using it without issue.  Got a couple questions for you: 

- Can you verify if you have any duplicate versions follow [these docs](https://docs.amplify.aws/lib/troubleshooting/upgrading/q/platform/js/#check-for-duplicate-versions)?
- What version of npm are you using? (run `npm -v` to see)

I'll see if we can find a workaround or way to unblock you.  As for the SDK upgrade, I've got no updates to provide at this time on when that will be released.

**Comment by stephenh** [comment]
@cwomack ah wow, thank you for the reply! Unfortunately I've been "sneaking in" our upgrade to React 18 in between other day-to-day tasks and haven't a chance to try your suggestions. Just wanted to say I really appreciate the reply, and will try your suggestions hopefully in the next few days/this weekend-ish.

**Comment by stephenh** [comment]
@cwomack erg... apologies but yes this is working for me now!

I didn't have dup versions of amplify, but I was using node 16 + whatever version of npm that was, 7.10.0, and so I a) bumped to Node 18.13.0 and b) also switched to yarn v3 and `yarn install` _immediately_ worked the first time.

Which is great! But yeah, sorry for the false bug report.

The nudge on "old npm" was what did it; thank you!

**Comment by cwomack** [comment]
@stephenh, no worries at all! Thank you for the reply and glad everything is working.  Feel free to follow up or open an new issue if need be should any other issues comes up.
